CMakeLists.txt

```diff
@@ -514,7 +527,13 @@ if(WITH_DOC)
 	# just switching off the Doxygen's HAVE_DOT option
 	find_package_handle_standard_args(Dot REQUIRED_VARS DOXYGEN_DOT_EXECUTABLE)
 
-	include(cmake/CheckCaseSensitiveFileSystem.cmake)
+
+        string(REPLACE ";" " " LIBIIO_DOXYGEN_DEFINES "${LIBIIO_DOXYGEN_DEFINES}")
+        string(REPLACE ";" " " DOXYGEN_ENABLED_SECTIONS "${DOXYGEN_ENABLED_SECTIONS}")
+        string(REPLACE ";" " " DOXYGEN_INPUT "${DOXYGEN_INPUT}")
+        string(REPLACE ";" " " DOXYGEN_STRIP_FROM_PATH "${DOXYGEN_STRIP_FROM_PATH}")
```

Comment:
Looking at the CMake script, those lists will only ever have one entry each, do you really need to use lists there? I would think that using regular CMake strings would be enough. Then you wouldn't have to do character substitution like that.

Reply:
`DOXYGEN_INPUT` can have multiple entries. As the others are conceptually lists as well they got the same treatment. Just for consistency.

Reply:
It would probably be better to replace your:
```cmake
list(APPEND DOXYGEN_INPUT "${CMAKE_SOURCE_DIR}/bindings/cpp/")
```
with:
```cmake
set(DOXYGEN_INPUT "${DOXYGEN_INPUT} ${CMAKE_SOURCE_DIR}/bindings/cpp/")
```

Then none of them would need character substitution.

Note that with CMake 3.x you could use `string(APPEND ...)` or `list(JOIN ...)` but we can't use them yet as we keep compatibility with CMake 2.8.

Reply:
Done.